spec/builders/latexmk-spec.js

```diff
@@ -214,5 +222,52 @@ describe('LatexmkBuilder', () => {
         expect(exitCode).toBe(11)
       })
     })
+
+    function doCustomRuleCheck (label, name, extensions, outputDirectory) {
+      it(label, () => {
+        let dir = path.join(fixturesPath, 'latexmk')
+        filePath = path.format({ dir, name, ext: '.tex' })
+        dir = path.join(dir, outputDirectory)
+        atom.config.set('latex.enableLatexmkrc', true)
+        atom.config.set('latex.outputDirectory', outputDirectory)
+
+        waitsForPromise(() => {
+          return builder.run(filePath).then(code => { exitCode = code })
+        })
+
+        runs(() => {
+          expect(exitCode).toBe(0)
+          for (const ext of extensions) {
+            const output = path.format({ dir, name, ext })
+            expect(fs.existsSync(output)).toBe(true, `Check the existance of ${ext} file.`)
+          }
+        })
+      })
+    }
+
+    function checkCustomRules (label, name, extensions) {
+      doCustomRuleCheck(label, name, extensions, '')
+      doCustomRuleCheck(`${label} with an output directory`, name, extensions, 'build')
+    }
+
+    checkCustomRules('successfully creates glossary files when using the glossaries package',
+      'glossaries-test',
+      ['.acn', '.acr', '.glo', '.gls', '.pdf']
+    )
+
+    checkCustomRules('successfully creates nomenclature files when using the nomencl package',
+      'nomencl-test',
+      ['.nlo', '.nls', '.pdf']
+    )
+
+    checkCustomRules('successfully creates index files when using the index package',
+      'index-test',
+      ['.idx', '.ind', '.ldx', '.lnd', '.pdf']
+    )
+
+    checkCustomRules('successfully creates SageTeX files when using the sagetex package',
+      'sage-test',
+      ['.sagetex.sage', '.sagetex.sout', '.pdf']
+    )
```

Comment:
Is there some way we can avoid all this "custom" stuff? It makes the tests really hard to read. Let's stick to more idiomatic Jasmine code. Basically I don't want to see any `it` calls inside other functions or in loops, etc.

I'd much rather have code duplication in specs, than lots of nested function calls etc which makes it hard to read the specs. If I have to "parse" the code to figure out what tests will be run, there's a problem.